File: app/src/main/java/by/nts/cafe/app/CafeApp.java
package by.nts.cafe.app;

import android.app.Application;
import android.preference.PreferenceManager;

import by.nts.cafe.app.dao.AppDatabase;
import by.nts.cafe.app.helper.rx.Transformers;
import io.reactivex.disposables.Disposable;

public class CafeApp extends Application {
    public static final String LOG = "cafe-nts-log";

    private static AppDatabase APP_DB;

    @Override
    public void onCreate() {
        super.onCreate();

        PreferenceManager.setDefaultValues(this, R.xml.pref, false);
        APP_DB = AppDatabase.buildDatabase(getApplicationContext());

        //initDBIfEmpty();
    }


    public void initDBIfEmpty() {
        Disposable d = getAppDatabase().hallDao().getAll()
                .doOnSuccess(list -> {
                    if (list.isEmpty()) {
                        AppDatabase.initTempData(getAppDatabase());
                    }
                })
                .compose(Transformers.schedulersIOMaybe())
                .subscribe();
    }

    public static AppDatabase getAppDatabase() {
        return APP_DB;
    }

}

File: app/src/main/java/by/nts/cafe/app/CafeApp.kt
package by.nts.cafe.app

import android.app.Application
import android.preference.PreferenceManager
import by.nts.cafe.app.dao.AppDatabase
import by.nts.cafe.app.helper.rx.attachTo
import by.nts.cafe.app.helper.rx.schedulersIOFlowable
import by.nts.cafe.app.locator.ServiceLocator
import by.nts.cafe.app.model.db.HallModel
import by.nts.cafe.app.presentation.presenter.PresenterFactory
import io.reactivex.disposables.CompositeDisposable

class CafeApp : Application() {
    companion object {
        const val LOG = "cafe-nts-log"

        lateinit var appDatabase: AppDatabase
            private set
    }

    override fun onCreate() {
        super.onCreate()

        PreferenceManager.setDefaultValues(this, R.xml.pref, false)
        appDatabase = AppDatabase.buildDatabase(applicationContext)

        PresenterFactory.instance.setContext(applicationContext)
        ServiceLocator.instance.setContext(applicationContext)


        initDBIfEmpty();
    }

    private fun initDBIfEmpty() {
        val compositeDisposable = CompositeDisposable()
        ServiceLocator.instance.getHallDao().all
                .doOnNext { list ->
                    if (list.isEmpty()) {
                        AppDatabase.initTempData(appDatabase)
                    }
                }
                .compose<List<HallModel>>(schedulersIOFlowable<List<HallModel>>())
                .subscribe { compositeDisposable.dispose() }
                .attachTo(compositeDisposable)
    }

}
